In [9]:
#imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import math
import random

Question 1

In [1]:
scores = [65, 70, 68, 73, 75, 80, 85, 78, 90, 100]

a.

In [3]:
mean = sum(scores) / len(scores)
mean

78.4

In [6]:
s = np.std(scores)
s

np.float64(10.229369482035537)

In [8]:
r = 2.262 * s / math.sqrt(len(scores))
print(mean - r)
print(mean + r)

71.08285828919516
85.71714171080485


The sample mean is 78.4, the standard deviation is 10.23, and the confidence interval is [71.08, 85.72].

b.

In [27]:
def bootstrap(data, n):
    means = []
    for j in range(n):
        indices = []
        bootstrap_sample = []
        for i in data:
            indices.append(random.randint(0,len(data)-1))
        for index in indices:
            bootstrap_sample.append(data[index])
        mean = sum(bootstrap_sample)/len(bootstrap_sample)
        means.append(mean)
    return means
    
means = bootstrap(scores, 40)
bootstrap_mean = sum(means) / len(means)
bootstrap_s = np.std(means)
bootstrap_r = 2.262 * bootstrap_s / math.sqrt(len(means))
print(f"The confidence interval is [{bootstrap_mean-bootstrap_r},{bootstrap_mean+bootstrap_r}]")

The confidence interval is [77.63058884849136,79.93941115150864]


c.

I would use the bootstrap method because the confidence interval that it gives, is much smaller, so we can be more sure of what population mean is.

Question 2.

a.